eras/babbage/impl/src/Cardano/Ledger/Babbage.hs

```diff
@@ -0,0 +1,281 @@
+              ( smartUTxOState
+                  initialUtxo
+                  (Coin 0)
+                  (Coin 0)
+                  def
+              )
+              (DPState (def {_genDelegs = GenDelegs genDelegs}) def)
+          )
+          (extendPPWithGenesis pp ag)
+          (extendPPWithGenesis pp ag)
+          def
+      )
+      SNothing
+      (PoolDistr Map.empty)
+    where
+      initialEpochNo = 0
+      initialUtxo = genesisUTxO sg
+      reserves =
+        coin $
+          inject (word64ToCoin (sgMaxLovelaceSupply sg))
+            <-> balance initialUtxo
+      genDelegs = sgGenDelegs sg
+      pp = sgProtocolParams sg
+
+instance CC.Crypto c => UsesTxOut (BabbageEra c) where
+  makeTxOut _proxy addr val = TxOut addr val NoDatum
+
+instance CC.Crypto c => API.CLI (BabbageEra c) where
+  evaluateMinFee = minfee
+
+  evaluateConsumed = consumed
+
+  addKeyWitnesses (ValidatedTx b ws aux iv) newWits = ValidatedTx b ws' aux iv
+    where
+      ws' = ws {txwitsVKey = Set.union newWits (txwitsVKey ws)}
+
+  evaluateMinLovelaceOutput pp out =
+    Coin $ utxoEntrySize out * unCoin (_coinsPerUTxOWord pp)
+
+type instance Core.Tx (BabbageEra c) = ValidatedTx (BabbageEra c)
+
+type instance Core.TxOut (BabbageEra c) = TxOut (BabbageEra c)
+
+type instance Core.TxBody (BabbageEra c) = TxBody (BabbageEra c)
+
+type instance Core.Value (BabbageEra c) = V.Value c
+
+type instance Core.Script (BabbageEra c) = Script (BabbageEra c)
+
+type instance Core.AuxiliaryData (BabbageEra c) = AuxiliaryData (BabbageEra c)
+
+type instance Core.PParams (BabbageEra c) = PParams (BabbageEra c)
+
+type instance Core.Witnesses (BabbageEra c) = TxWitness (BabbageEra c)
+
+type instance Core.PParamsDelta (BabbageEra c) = PParamsUpdate (BabbageEra c)
+
+instance CC.Crypto c => UsesValue (BabbageEra c)
+
+instance (CC.Crypto c) => UsesPParams (BabbageEra c) where
+  mergePPUpdates _ = updatePParams
+
+instance CC.Crypto c => ValidateAuxiliaryData (BabbageEra c) c where
+  hashAuxiliaryData x = AuxiliaryDataHash (hashAnnotated x)
+  validateAuxiliaryData (AuxiliaryData metadata scrips) =
+    all validMetadatum metadata
+      && all validScript scrips
+
+instance CC.Crypto c => EraModule.SupportsSegWit (BabbageEra c) where
+  type TxSeq (BabbageEra c) = Alonzo.TxSeq (BabbageEra c)
+  fromTxSeq = Alonzo.txSeqTxns
+  toTxSeq = Alonzo.TxSeq
+  hashTxSeq = Alonzo.hashTxSeq
+  numSegComponents = 4
+
+instance API.ShelleyEraCrypto c => API.ShelleyBasedEra (BabbageEra c)
+
+-------------------------------------------------------------------------------
+-- Era Mapping
+-------------------------------------------------------------------------------
+
+-- Rules inherited from Alonzo
+
+type instance Core.EraRule "UTXOS" (BabbageEra c) = Alonzo.UTXOS (BabbageEra c)
+
+type instance Core.EraRule "UTXO" (BabbageEra c) = Alonzo.AlonzoUTXO (BabbageEra c)
+
+type instance Core.EraRule "UTXOW" (BabbageEra c) = Alonzo.AlonzoUTXOW (BabbageEra c)
```

Comment:
let's leave a TODO here to map the three UTXO rules to the new Babbage ones once they exist.